### ISSUE TITLE
Upstream numenta/NAB now supports py3, how to merge? 

### ISSUE THREAD

**Issue by breznak** [htm-community/NAB#21]
I am wondering how/if we would merge with upstream NAB from Numenta, which is now Python3 too. 

This codebase has been converted a long time before that by @ctrl-z-9000-times , and a bunch of other work has been done atop of that. 

What do you think? Attempt merging, start anew on theirs master and add our changes, or keep this as diverged fork?

**Comment by D-McDougall** [comment]
Well I looked through the diff's between ours and Numenta's NAB, and there are a lot of changes... It appears that they are not 100% python3, which is disappointing given that there is a program which automatically converts py2 to py3.

IMO we should merge our work into theirs, and then drop this repo.
Getting htm.core to work with NAB is the really important thing.

However I don't have the effort to work on this project anymore.
Best of luck,
dmac

**Comment by breznak** [comment]
> IMO we should merge our work into theirs, and then drop this repo.

I'll try to get input how open Numenta would be to merge (parts?) of this work. 
I agree a single, unified NAB repo would be ideal. 

> Getting htm.core to work with NAB is the really important thing.

yep. So far NAB is the most important use-case/validation for the changes we make in the wild. 
Unfortunately, #15 but I should work on that. 

> However I don't have the effort to work on this project anymore.
Best of luck,

Thanks for all you work David, been pleasure working with you, I love your research ideas (that are now mostly part of HTM.core). 

Should I still CC you on normal basis, or you are (hope temporarily?) off the loop and @ctrl-z-9000-times you only if needed urgently? 

Take care and best of luck to you too,

**Comment by D-McDougall** [comment]
> Should I still CC you on normal basis

Of course, but I probably won't be doing any code reviews.  I have new projects which I'm devoting my attention to.  Also, I still check the forum regularly.

**Comment by breznak** [comment]
sure, that's fine. Great that you'll be reachable on the forums or for insight if needed :+1:

**Comment by rhyolight** [comment]
> IMO we should merge our work into theirs, and then drop this repo.

Our company priorities are not on NAB right now. There are only a couple people here that can review these PRs and they are fully engaged. I'm just warning that if you created PRs against NAB, I cannot promise when or if they will be merged.

**Comment by breznak** [comment]
>  I'm just warning that if you created PRs against NAB, I cannot promise when or if they will be merged.

thanks Matt, that;s valuable information. In that case I'd (for now) keep using the community NAB, only question: 
- try to merge Numenta's changes and place ours on top of that?
- or just keep rocking as is with this repo (keep an eye on fixes upstream)

**Comment by psteinroe** [comment]
I don't think I have the credibility to decide that but especially since Numenta focusses on other topics and their code seems to be a mix of py2 and py3 I would keep rocking with this repo here while keeping an eye on upstream to ensure that it only differs implementation-wise and not concept-wise.
